tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Lambda/HandlerWrapperSetHandlerIntegration.cs

```diff
@@ -108,18 +110,18 @@ public async Task<TInnerReturn> OnDelegateEndAsync<TInnerReturn>(object sender,
                 if (proxyInstance == null)
                 {
                     LambdaCommon.Log("DuckCast.IInvocationResponse got null proxyInstance", debug: false);
-                    await LambdaCommon.EndInvocationAsync(string.Empty, exception, ((CallTargetState)state!).Scope, RequestBuilder).ConfigureAwait(false);
+                    await LambdaCommon.EndInvocationAsync(string.Empty, exception, ((CallTargetState)state!).Scope, ((CallTargetState)state!).State, RequestBuilder).ConfigureAwait(false);
```

Comment:
To minimize all the casting (`((CallTargetState)state!)`) in this method, you can do it just once. Maybe also rename the method parameter from `state` to `stateObject` so we can use `state` below:
```csharp
public async Task<TInnerReturn> OnDelegateEndAsync<TInnerReturn>(object sender, TInnerReturn returnValue, Exception exception, object stateObject)
{
    var state = (CallTargetState)state!;
    ...
```

Then you can simply use `state.State` here and `state.Scope` below.